packaging/debian/postinst

```diff
@@ -25,31 +25,70 @@ DAEMON_USER=storj-storagenode
 
 . /usr/share/debconf/confmodule
 
+get_binary() {
+    binary=$1
+    tmpfile=$(mktemp /tmp/${binary}.zip.XXXXXX)
+    wget -O $tmpfile ${BINARIES_SERVER}/${binary}_linux_amd64.zip
+    unzip $tmpfile -d /var/lib/storagenode
+    mv /var/lib/storagenode/${binary}_linux_amd64 /var/lib/storagenode/${binary}
+    rm $tmpfile
+}
 
 case "$1" in
     configure)
 	installed_version="$2"
-	#if [ $installed_version != "" ]; then
-	#    echo "This is a reconfiguration or a reinstallation"
-	#fi
+	if [ ! -z "$installed_version" ]; then
+	    echo "This is a reconfiguration or a reinstallation"
+	fi
+
+	db_get storagenode/identitydir
+	identitydir="$RET"
+	db_get storagenode/wallet
+	wallet="$RET"
+	db_get storagenode/email
+	email="$RET"
+	db_get storagenode/external_addr
+	external_addr="$RET"
+	db_get storagenode/storagedir
+	storagedir="$RET"
+	db_get storagenode/diskspace
+	diskspace="$RET"
 
-	# TODO: should not download binaries if it's only a reconfigure
-	#echo "Downloading binaries..."
 	for binary in "storagenode" "storagenode-updater"; do
-	    wget -O /tmp/${binary}.zip ${BINARIES_SERVER}/${binary}_linux_amd64.zip
-	    unzip /tmp/${binary}.zip -d /var/storagenode/
-	    mv /var/storagenode/${binary}_linux_amd64 /var/storagenode/${binary}
-	    # so that we're in PATH easily
-	    ln -s /var/storagenode/${binary} /usr/bin/${binary}
+	    if [ ! -f "/var/lib/storagenode/${binary}" ]; then
+	        echo "Downloading ${binary} binary..."
+	        get_binary $binary
+	    fi
 	done
-	adduser --system --group --no-create-home --quiet $DAEMON_USER
 
-#	if [ "$old_storagedir" != "" ] && [ "$storagedir" != "$old_storagedir" ]; then
-#	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
-#	else
-#	    echo "Storage directory: ${storagedir}"
-#	fi
+	if [ ! -z "$old_storagedir" ] && [ "$storagedir" != "$old_storagedir" ]]; then
+	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
+	else
+	    echo "Storage directory: ${storagedir}"
+	fi
+
+	# storagenode setup
+	if [ ! -f "/etc/storagenode/config.yaml" ]; then
+	    /var/lib/storagenode/storagenode setup \
+	    --config-dir "/etc/storagenode" \
+	    --identity-dir "$identitydir" \
+	    --operator.email "$email" \
+	    --operator.wallet "$wallet" \
+	    --contact.external-address "$external_addr" \
+	    --storage.path "$storagedir" \
+	    --storage.allocated-disk-space "$diskspace"
+
+	    # fix permissions
+	    chmod 644 /etc/storagenode/config.yaml
```

Comment:
When trying it, I have a permission problem. The identity directory is not owned by the daemon user, so the service does not start. I don't know what's the best options to deal with this..

Reply:
we can copy identity during installation to `/etc/storagenode/identity`, or we can change group for the provided `identity` folder I don't know which is better approach.

Reply:
@thepaul what do you think?

Reply:
I'm not sure the storagenode _should_ need write access to the identity directory. Do we know why it checks for that?

Reply:
it does not. But it should have the read access to the identity directory.

Reply:
@thepaul I don't think it's needed, it seems @Fadila82 identity folder has no read access for other users. And here is the main question, should the deb expects user to have their identity in a folder that is accessible for everyone(Read access)

Reply:
Oh I see. If we're talking about the identity directory provided by the user in debconf, then I think we should probably copy it somewhere. `/etc/storagenode/identity` sounds fine.

We definitely don't want to change the group on the user's provided directory. They won't expect that.